data_node/upload_manager.go

```diff
@@ -116,9 +116,14 @@ func (um *UploadManager) handleAppendUpload(w http.ResponseWriter, r *http.Reque
 	r.Body = http.MaxBytesReader(w, r.Body, um.maxChunkSize)
 	log.Println(um.logPrefix, r.RemoteAddr, "Received APPEND request")
 
+	if r.ContentLength > um.maxChunkSize {
+		log.Println(um.logPrefix, r.RemoteAddr, "Request body too large")
+		handleRequestError(w, http.StatusBadRequest, fmt.Sprintf("Maximum allowed content length is %d", um.maxChunkSize))
+		return
+	}
+
```

Comment:
Can we move this check at the very top before reading the truncated body? Also, are you aware of a more robust way? I mean what if the request is large but the content length header was altered?